common/utils.py

```diff
@@ -81,3 +86,39 @@ def get_client_ip_address(request: HttpRequest) -> IPAddressString | None:
         return None
     else:
         return IPAddressString(client_ip)
+
+
+def ip_address_check(function):
```

Comment:
Could you please add a doc comment here that explains what this decorator does? It's pretty magical :) In particular, we should document how it interacts with a parameter named `assignment_id`.